File: js-packages/admin-ui/src/openapi-generated/models/EnrichItemDTO.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { BehaviorMergeType } from './BehaviorMergeType';
import type { BehaviorOnError } from './BehaviorOnError';
import type { EnrichItemType } from './EnrichItemType';
import type { ResourceUri } from './ResourceUri';

export type EnrichItemDTO = {
    name: string;
    description?: string;
    type: EnrichItemType;
    resourceUri: ResourceUri;
    script?: string;
    jsonConfig?: string;
    jsonPath: string;
    behaviorMergeType: BehaviorMergeType;
    requestTimeout: number;
    behaviorOnError: BehaviorOnError;
};

File: core/app/datasource/src/main/java/io/openk9/datasource/pipeline/actor/dto/EnrichItemDTO.java
package io.openk9.datasource.pipeline.actor.dto;

import io.openk9.datasource.model.EnrichItem;
import lombok.Data;

@Data
public class EnrichItemDTO {
	private Long id;
	private String name;
	private String serviceName;
	private String script;
	private EnrichItem.EnrichItemType type;
	private String jsonConfig;
	private String jsonPath;
	private EnrichItem.BehaviorMergeType behaviorMergeType;
	private Long requestTimeout;
	private EnrichItem.BehaviorOnError behaviorOnError;
}
